src/psyclone/psyGen.py

```diff
@@ -976,9 +976,39 @@ def __init__(self, ast=None, children=None, parent=None):
         self._ast_end = None
         # List of tags that provide additional information about this Node.
         self._annotations = []
+        # Name to use for this Node type. By default we use the name of
+        # the class but this can be overridden by a sub-class.
+        self._text_name = self.__class__.__name__
+        # Which colour to use from the SCHEDULE_COLOUR_MAP
+        self._colour_key = self.__class__.__name__
+
+    def coloured_name(self, colour=True):
+        '''
+        :returns: the name of this node, optionally with colour control codes \
+                  (requires that the termcolor package be installed).
+        :rtype: str
+        '''
+        if colour:
+            try:
+                return colored(self._text_name,
+                               SCHEDULE_COLOUR_MAP[self._colour_key])
+            except KeyError:
+                pass
+        return self._text_name
+
+    def node_str(self, colour=True):
+        '''
+        :param bool colour: whether or not to include control codes for \
+                            coloured text.
+
+        :returns: a text description of this node. Will typically be \
+                  overridden by sub-class.
+        :rtype: str
+        '''
+        return self.coloured_name(colour)
```

Comment:
Oh yes, that's better. Note that I had to change several of the `node_str` methods to call `self.coloured_name` rather than `super(xxx, self).node_str`.